### PR TITLE
Migrate to 1ES templates for internal builds

### PR DESCRIPTION
**Pull request by dsplaisted**
Switch to 1ES pipeline templates for internal builds.

Internal PR with these changes to test the pipeline changes: https://dev.azure.com/dnceng/internal/_git/dotnet-sdk/pullrequest/38177

This PR follows the process outlined in this dotnet/installer PR: https://github.com/dotnet/installer/pull/19016

Additional notes:

- `matrix:` elements were replaced.  In most cases there weren't actually multiple value sets, so the matrix could just be removed.  For dotnet-format, the matrix of repos to test on was replaced with a template parameter with default values and an `each` preprocessor loop
- template parameter variables had to change syntax
- For template engine yaml, copied it directly into the internal .vsts-ci.yml file
- For dotnet-format, updated the yaml template to work for both internal and external builds to avoid having to duplicate it

### CODE REVIEW

**Comment by joeloff** [comment]
Template Engine tests have an odd failure:

`Assert.Equal() Failure: Strings differ\n           ↓ (pos 0)\nExpected: "Console Applica..."\nActual:   "BLiP Console (...."\n           ↑ (pos 0)`

**Comment by baronfel** [comment]
@joeloff that suggests to me that there's a different template being invoked than we expected - did the `console` shortname get clobbered or something?

**Comment by joeloff** [comment]
> @joeloff that suggests to me that there's a different template being invoked than we expected - did the `console` shortname get clobbered or something?

I'm not sure, that would be weird to have the YAML influence template properties

**Comment by joeloff** [comment]
Maybe someone published a new template so search output is no longer valid in the test?

**Comment by baronfel** [comment]
That's what I'm thinking. Especially because we just updated the search index with Noah's changes yesterday

**Comment by marcpopMSFT** [comment]
Likely the test needs a baseline update. I believe @dotnet/domestic-cat is already looking into it.

**Comment by nagilson** [comment]
> Likely the test needs a baseline update. 

What's that?

File: .vsts-ci.yml [hunk 1]
```diff
@@ -47,164 +47,286 @@ variables:
...
-        name: 'Managed'
-        container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8'
-  - ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
-    - template: /eng/build.yml
-      parameters:
-        agentOs: Windows_NT_FullFramework
+        image: 1es-windows-2022
+        os: windows
+    customBuildTags:
+    - ES365AIMigrationTooling
```
Comment (MiYanni) [inline .vsts-ci.yml:68]: We don't need this. It is tracking information for using the conversion tool, which we didn't use here.

File: .vsts-ci.yml [hunk 2]
```diff
@@ -47,164 +47,286 @@ variables:
...
-        matrix:
-          Build_Release:
-            _BuildConfig: Release
-            _PublishArgs: '-publish /p:DotNetPublishUsingPipelines=true'
-            ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
-              _SignType: test
-              _Test: -test
-            ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-              _SignType: real
-              _Test: ''
```
Comment (MiYanni) [inline .vsts-ci.yml:91]: So, after testing and more discussions, we learned we didn't actually need to change this. Only `strategy` that applied to `pool` information needed to be changed. But you can leave the changes here as it really had no reason to use a `strategy` to begin with.

File: .vsts-ci.yml [hunk 3]
```diff
@@ -47,164 +47,286 @@ variables:
...
+            pool:
+              ${{ if eq(variables['System.TeamProject'], 'public') }}:
+                name: $(DncEngPublicBuildPool)
+                image: 1es-windows-2022-open
+                os: windows
+              ${{ if ne(variables['System.TeamProject'], 'public') }}:
+                name: $(DncEngInternalBuildPool)
+                # Should this be windows.vs2022preview.amd64?
+                image: 1es-windows-2022
+                os: windows
```
Comment (MiYanni) [inline .vsts-ci.yml:138]: If you go to https://helix.dot.net/ and find `windows.vs2022preview.amd64`, you can click on it and it'll show you the details of the image. However, if you also find `1es-windows-2022` and click on it, it doesn't have any details. So... ❓❓❓

**Review by MiYanni — APPROVED** [review]

File: eng/build.yml [hunk 4]
```diff
@@ -40,22 +40,35 @@ jobs:
...
+          - ${{ if ne(variable.name, '') }}:
+            - name: ${{ variable.name }}
+              value: ${{ variable.value }}
+        - name: _AgentOSName
+          value: ${{ parameters.agentOs }}
+        - name: _TeamName
+          value: DotNetCore
+        - name: _OfficialBuildIdArgs
+          value: ''
+        - name: _PublishArgs
```
Comment (mmitche) [inline eng/build.yml:53]: I think you're overriding the publish args with an empty set in the official build.

**Pull request merged**